src/compiler/checker.ts

```diff
@@ -2094,7 +2106,7 @@ module ts {
             }
             // Use type from type annotation if one is present
             if (declaration.type) {
-                return getTypeFromTypeNode(declaration.type);
+                return getTypeFromTypeNodeOrHeritageClauseElement(declaration.type);

```

Comment:
I really do not like this name. why do we need it here? why can not we just have getTypeOfNode?